starlite/contrib/htmx/response.py

```diff
@@ -0,0 +1,157 @@
+import json
+from typing import Any, Dict, Literal, TypeVar
+from urllib.parse import quote
+
+from starlite import Response, Template
+from starlite.status_codes import HTTP_200_OK
+
+EventAfterType = Literal["receive", "settle", "swap"]
+
+# HTMX defined HTTP status code.
+# Response carrying this status code will ask client to stop Polling.
+HTMX_STOP_POLLING = 286
+T = TypeVar("T")
+
+
+class HXStopPolling(Response):
+    """Stop HTMX client from Polling."""
+
+    def __init__(self) -> None:
+        """Initialize"""
+        super().__init__(content=None)
+        self.status_code = HTMX_STOP_POLLING
+
+
+class ClientRedirect(Response):
+    """HTMX Response class to support client side redirect."""
+
+    def __init__(self, url: str) -> None:
+        """Set status code to 200 (required by HTMX),
+        and pass redirect url
+        """
+        super().__init__(
+            content=None,
+            status_code=HTTP_200_OK,
+            headers={"HX-Redirect": quote(url, safe="/#%[]=:;$&()+,!?*@'~"), "Location": ""},
+        )
+        del self.headers["Location"]
+
+
+class ClientRefresh(Response):
+    """Class to support HTMX client page refresh"""
+
+    def __init__(self) -> None:
+        """Set Status code to 200 and set headers."""
+        super().__init__(content=None, status_code=HTTP_200_OK, headers={"HX-Refresh": "true"})
+
+
+class PushUrl(Response):
+    """Class to push new url into the history stack"""
+
+    def __init__(self, content: T, url: str = "", **kwargs: Any) -> None:

```

Comment:
`T` is not bound. You have to bind it to `Generic` and pass it to Response:

```python
class PushUrl(Generic[T], Response[T])
```

This applies everywhere you use a type-var.